core/src/main/java/hudson/FilePath.java

```diff
@@ -3296,10 +3297,11 @@ public Boolean invoke(@Nonnull File parentFile, @Nonnull VirtualChannel channel)
                 try{
                     Path child = currentFileAbsolutePath.toRealPath();
                     if (!child.startsWith(parentRealPath)) {
+                        LOGGER.log(Level.FINE, "Child [{0}] does not start with parent [{1}] => not descendant", new Object[]{ child, parentRealPath });
                         return false;
                     }
                 } catch (NoSuchFileException e) {
-                    // nonexistent file
+                    // nonexistent file / Windows Server 2016 + MSFT docker
```

Comment:
I guess this is not the case anymore?

Reply:
`toRealPath` will never work under Windows Server 2016 + MSFT Docker (perhaps other env) when the root is a mounted folder.